queries.sql

```diff
@@ -8,3 +8,32 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3 OR (weight_kg = 10.4 OR weight_kg = 17.3);
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT DEL2022;
+UPDATE animals SET weight_kg = weight_kg * (-1);
+ROLLBACK TO DEL2022;
+UPDATE animals SET weight_kg = weight_kg * (-1) WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) as average FROM animals;
+SELECT neutered, SUM(escape_attempts) as max_escape_attempts FROM animals GROUP BY neutered ORDER BY max_escape_attempts DESC LIMIT 1;
```

Comment:
- [ ] Please note that we want to know if it was the neutered animals who escaped the most or the not neutered animals. That can be achieved by getting the average of escapes attempted per neutered and not neutered. Therefore you code could look like this :point_down: 
```sql
   SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```
or 
```sql
  SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```